### DISCUSSION TITLE
Slow page loads on MacOS Safari

### DISCUSSION THREAD

**Discussion by CelticWebSolutions** [netbox-community/netbox#16447] [Help Wanted!]
### Deployment Type

Self-hosted

### NetBox Version

v4.0.5

### Python Version

3.12

### Steps to Reproduce

Load a page with a large number of items, such as Tennants. in my case there are 88 items on the page

### Expected Behavior

Page loads promptly and all Tennants show and can be scrolled.

### Observed Behavior

On a browser such as Chrome, this works absolutely fine.  On macOS Safari, the page takes time to load after first being clicked, then when the page displays, it will have a single page full of content, scrolling results in it freezing and no additional content being shown, just blank page. After a period of around 10 - 15 seconds, then the rest of the content loads.

This has been the same since V4.0 but was not an issue prior to update.

**Comment by jeffgdotorg** [comment]
An otherwise empty instance with 100 Tenant objects in it rendered the full list in a sub-second timeframe.
![image](https://github.com/netbox-community/netbox/assets/1209268/c2026129-696d-4639-88db-99aafddb6722)

Might there be an interaction with other objects in your system?

**Comment by jeffgdotorg** [comment]
![image](https://github.com/netbox-community/netbox/assets/1209268/e435757e-1d27-495b-a8da-4da2563c2b2a)

Safari Version 17.5 (19618.2.12.11.6)

**Comment by jeffgdotorg** [comment]
Converting to a discussion since an actual bug seems unlikely. If that road leads back to a likely bug, we'll regroup in issues.

BTW thanks for reporting a problem you're having with NetBox, we do appreciate it.

**Comment by CelticWebSolutions** [comment]
Sorry for the delayed response, my Mac often runs dev software, so I'd already thought of this and tried it on my wife's and a friends Mac.  I get exactly the same issue I'm afraid.  

Before stepping up to V4, it was all running fine. I'm assuming its' something to do with the content in the Database,  originally I thought it was because the main system it ran on was running an old version of Debian, so we actually decided to fire up a new VM, install fresh Debain and transfer across, as you can see, that didn't make any difference either. 

We've upgraded through each iteration of V4, hoping it would fix the issue, but it has remained. I'm guessing far too much changed between V3 and V4 for it to be a simple find for whats the culprit of my issue.

Is there some kind of debug mode I can enable which will show what it's doing?

**Comment by alehaa** [comment]
I can't give steps to reproduce for this issue, but observe similar issues running 4.0.3. Also, sometimes I switch tabs and when coming back to NetBox the page is all white.

**Comment by bb-Ricardo** [comment]
Hi,

can also confirm the same issue. Chrome is loading and rendering page super fast. Safari is quite slow. Also scrolling on a page is slow. This issue exists since version 4.0.0.

Somehow the (re)rendering of the page causes the "Safari Graphics and Media" process to consume a lot of resources. The same happens when scrolling up and down.

<img width="470" alt="image" src="https://github.com/netbox-community/netbox/assets/771836/0b4e295b-1f36-4c59-914e-e8fd85d15a79">


System infos:

<img width="396" alt="image" src="https://github.com/netbox-community/netbox/assets/771836/427769f4-7582-47fc-b586-1f956fb31aa8">
<br>
<img width="262" alt="image" src="https://github.com/netbox-community/netbox/assets/771836/9d19f019-3214-4af6-9706-83d2af742114">

**Comment by mitchplze** [comment]
Seeing the same issues here. 

NetBox is almost completely unusable in Safari 17.5 on a brand new MacBook Pro with M3 and 16GB of RAM.

The screen tears and freezes scrolling up or down on any table. I have to set my row limit to 25 to even use it.

The exact same instance works flawlessly and lightning speed on Chrome 125, with 500 rows, on the same machine.

**Comment by CelticWebSolutions** [comment]
Seems there is some sort of bug after all then if multiple people are suffering from it. No data was changed as far as entries are concerned between V3 and V4 for me, data was updated of course, but no items added.  So it's something to do with the way these tables are generated in V4 when compared to V3. @jeffgdotorg, if it's of any use to anyone, I'm happy to do a remote session or run debugging for you to see if anything stands out for you.

**Comment by mitchplze** [comment]
I can't add much to this discussion, as I'm a new NetBox user who started with v4. But I can confirm the issue exists on two separate and unrelated new installations (both Docker), and seems to affect all Safari users: iOS and macOS.

But also happy to test / help if possible.

**Comment by Kage1** [comment]
Not sure if this is related, #16684, but several of us are seeing very poor performance issues after upgrading to 4.0.5. Most of us are not using Safari.

**Comment by jeffgdotorg** [comment]
We released NetBox 4.0.6 yesterday, with a fix for #16684. Anybody who's been able to reproduce this problem is invited to give it a try and weigh in on your experience.

**Reply by CelticWebSolutions** [reply]
Unfortunately it hasn't fixed the issue, updated today and I still have the same issue, it's possibly a little quicker, but still doesn't load pages properly unless left for a while.

**Comment by CelticWebSolutions** [comment]
Any updates on this yet?  It was moved from bugs/issues to discussions so I'm wondering if it's actually being looked in to?

**Comment by AliMickey** [comment]
Can confirm this is also an issue for us, especially on pages with tabled data such as interfaces it will load the base html but the content takes up to 10 seconds to render, tabbing out and back in will cause the same issue again.

**Comment by ognjen011** [comment]
This has been a problem since version v4.0. I have updated to v4.0.8 today and still the problem persists. None of the pages where you have to load more than 25 items are loading.  Also if the page does load it ends up unusable because as soon as you start to scroll it starts reloading and it gets stuck. I am using on the same machine Brave browser and it works just fine.

Some more information: as soon as i start loading the page in safari i noticed the Safari and Graphic Media process which averages around 1.5% cpu utilisation jumps instantly to 167% and sometimes up to 207%. This lasts a few seconds and then it goes down but the page does not load and as soon as i start scrolling again the process jumps backup to over 150%. I have disabled all extensions for the this tab and behaviour is the same.

**Comment by AliMickey** [comment]
@jeffgdotorg Can we get this back into an issue please? :) 

Happy to provide a video demonstration and logs if that's required.

**Comment by ZPrimed** [comment]
Not meant to be a silly "me too," but more of a further confirmation that this is definitely something within Safari's rendering process choking on NetBox pages when they have "a lot" of table entries. The "Safari Graphics and Media" sub-process just pegs one or more CPU cores during page load, and then again if you try to scroll too much. 

I'm on an M1 Macbook Pro, so it's not Intel-specific. Safari is not my "regular" browser, or else I'd be tearing my hair out over this.

This feels like it might be something that should be reported to Apple, since Safari is the only thing choking...? I have no problems with Vivaldi on the same system (which is Chromium-based).

**Comment by jeremystretch** [comment]
All of the evidence shared thus far suggests that this is a bug in Safari, as it is the only browser which experiences the issue. If anyone would like to propose a specific change to NetBox that they can confirm works around the issue with Safari, I'm happy to explore it.

**Reply by CelticWebSolutions** [reply]
I'm happy to report that this appears to be fixed in v4.10 :) 

I confirmed this by rolling back to 4.06, issue existed, upgraded again back to V4.1 and it is back to scrolling normally :)

So for all of you that were having an issue, get yourselves upgraded and see if it fixes it for you too :)

**Reply by bb-Ricardo** [reply]
can confirm, just moved from 4.0.6 to 4.1.1 and now it runs smoothly in Safari as well.

**Reply by jeremystretch** [reply]
FYI we had to hack the Tabler menu dropdown styling to remove a piece of (valid) CSS that was causing Safari to hang up. This is indeed a bug in Safari as suspected, but we implemented a workaround.

**Reply by bb-Ricardo** [reply]
Thank you for taking the initative and despite not your fault, implementing a workaround.
